calyx-py/calyx/builder.py

```diff
@@ -734,7 +734,7 @@ def bitwise_flip_reg(self, reg, cellname=None):
 
     def incr(self, reg, val=1, signed=False, cellname=None, static=False):
         """Inserts wiring into `self` to perform `reg := reg + val`."""
-        cellname = cellname or f"{reg.name}_incr"
+        cellname = cellname or f"{reg.name}_incr_{val}"
```

Comment:
This strategy is not long-term viable. I think we should do what `ir::Component` does and use a https://github.com/calyxir/calyx/blob/6f895a1353020ce254860c3aa0fcfa2ba1abf4c4/calyx-utils/src/namegenerator.rs#L7) like object which tracks all the defined names and ensures that we always generate unique names.

Reply:
The eDSL currently does a discount version of this: it maintains a counter starting at 0, and every time it makes a new name, it does something like `{the_op_in_question}_{counter++}`. This is why end up with things like `reg_0`, `add_1`, `mult_2`, etc. The true solution that you are suggesting would be really cool, but I'll park that as a separate nice-to-have issue if that's okay. In an in-person conversation with @polybeandip I asked him to move on for now using our discount solution.

Reply:
Yup, perfectly reasonable!

Reply:
This last commit should be more inline with the eDSL's current way to generate names (since we call `ComponentBuilder.generate_name` now).

This way, code like
```
a_incr_2 = comp.incr(a, 2)
a_incr_2_again = comp.incr(a, 2)
```
shouldn't raise errors anymore